hrqb/utils/__init__.py

```diff
@@ -31,6 +32,31 @@ def normalize_date(date: str | datetime.datetime) -> str | None:
     return None
 
 
+def convert_dataframe_columns_to_dates(
+    df: pd.DataFrame, columns: list[str]
+) -> pd.DataFrame:
+    """Convert select columns from a dataframe to datetime objects.
+
+    This more manual approach avoids a pandas error with pd.to_datetime() when the date
+    exceeds 2262-04-11.  Normally this would not be a problem, but employee appointments
+    that are ongoing receive a datetime of 2999-12-31. See: https://pandas.pydata.org/
+    pandas-docs/stable/user_guide/timeseries.html#timestamp-limitations.
+    """
+
+    def convert_to_date(
+        value: Any,  # noqa: ANN401
```

Comment:
Would it be possible to do `str | int | datetime.datetime | pd.Timestamp`?

Reply:
That's a great call.  I've updated to:

```python
value: str | datetime.datetime | pd.Timestamp,
```

The advantage here is that these are the _expected_ types, which will get handled by logic arms, but if we get an oddball type, it'll just return `None`.  In that way, the type hinting helps understand what it can/should handle.  Thanks for the suggestion!